src/main/java/it/gov/pagopa/pu/debtpositions/service/DebtPositionUpdateService.java

```diff
@@ -0,0 +1,151 @@
+import it.gov.pagopa.pu.debtpositions.model.PaymentOption;
+import it.gov.pagopa.pu.debtpositions.model.Transfer;
+import it.gov.pagopa.pu.debtpositions.repository.*;
+import jakarta.transaction.Transactional;
+import org.springframework.stereotype.Service;
+
+import java.util.Objects;
+
+@Service
+public class DebtPositionUpdateService {
+
+  private final DebtPositionRepository debtPositionRepository;
+  private final PaymentOptionRepository paymentOptionRepository;
+  private final InstallmentPIIRepository installmentRepository;
+  private final InstallmentNoPIIRepository installmentNoPIIRepository;
+  private final TransferRepository transferRepository;
+  private final DebtPositionMapper debtPositionMapper;
+
+  public DebtPositionUpdateService(DebtPositionRepository debtPositionRepository, PaymentOptionRepository paymentOptionRepository,
+                                   InstallmentPIIRepository installmentRepository, InstallmentNoPIIRepository installmentNoPIIRepository, TransferRepository transferRepository,
+                                   DebtPositionMapper debtPositionMapper) {
+    this.debtPositionRepository = debtPositionRepository;
+    this.paymentOptionRepository = paymentOptionRepository;
+    this.installmentRepository = installmentRepository;
+    this.installmentNoPIIRepository = installmentNoPIIRepository;
+    this.transferRepository = transferRepository;
+    this.debtPositionMapper = debtPositionMapper;
+  }
+
+  @Transactional
+  public void updateDebtPositionFromDTO(DebtPosition debtPosition, DebtPositionDTO debtPositionDTO) {
+    debtPositionMapper.updateFromDTO(debtPosition, debtPositionDTO);
+    DebtPosition updatedDebtPosition = debtPositionRepository.save(debtPosition);
+    alignDebtPositionDTO(debtPositionDTO, updatedDebtPosition);
+
+    debtPosition.getPaymentOptions().forEach(paymentOption -> {
+      paymentOption.setDebtPositionId(updatedDebtPosition.getDebtPositionId());
+      PaymentOption savedPaymentOption = paymentOptionRepository.save(paymentOption);
+      PaymentOptionDTO paymentOptionDTO = alignPaymentOptionDTO(debtPositionDTO, savedPaymentOption);
+
+      paymentOption.getInstallments().forEach(installmentNoPII -> {
+        InstallmentDTO installmentDTO = paymentOptionDTO.getInstallments().stream()
+          .filter(i -> Objects.equals(i.getIud(), installmentNoPII.getIud()))
+          .findFirst().orElseThrow();
+
+        installmentDTO.setPaymentOptionId(savedPaymentOption.getPaymentOptionId());
+        InstallmentNoPII savedInstallment = installmentRepository.save(installmentDTO).getNoPII();
+        alignInstallmentDTO(installmentDTO, savedInstallment);
+
+        installmentNoPII.getTransfers().forEach(transfer -> {
+          transfer.setInstallmentId(savedInstallment.getInstallmentId());
+          Transfer savedTransfer = transferRepository.save(transfer);
+          alignTransferDTO(installmentDTO, savedTransfer);
+        });
+      });
+    });
```

Comment:
I see dangerous to have all these lines copied from it.gov.pagopa.pu.debtpositions.service.DebtPositionSaveService#saveDebtPositionDTO

why don't you copy the original id from model to DTO and call directly the save operation on the DTO?